### PR TITLE
[OPS-6907] FML

### PR DESCRIPTION
**Pull request by cafuego**
This PR also includes previously unsuccessful hacky work to try and get thread ID support into the plugin. Sadly the API calls it use seem to not support that. Maybe a thing to try and fix in a year or two.

### CODE REVIEW

File: src/main/java/com/flowdock/jenkins/ChatMessage.java [hunk 1]
```diff
@@ -72,4 +76,38 @@ else if(build.getResult() == Result.NOT_BUILT) {
...
+        } else {
+            projectName = build.getProject().getDisplayName();
+        }
+
+        String rootUrl = Hudson.getInstance().getRootUrl();
+        String buildLink = (rootUrl == null) ? null : rootUrl + build.getUrl();
+        boolean hasLink = buildLink != null;
+        String buildNo = build.getDisplayName().replaceAll("#", "");
+
+        content.append(":rocket:");
```
Comment (emmajane) [inline src/main/java/com/flowdock/jenkins/ChatMessage.java:99]: You had me at rockets.

File: src/main/java/com/flowdock/jenkins/ChatMessage.java [hunk 2]
```diff
@@ -72,4 +76,38 @@ else if(build.getResult() == Result.NOT_BUILT) {
...
+        if(hasLink) {
+            content.append("[");
+        }
+        content.append(projectName + configuration).append(" build ").append(buildNo);
+        content.append(" **is starting**");
+        if(hasLink) {
+            content.append("]");
+            content.append("(" + buildLink + ")");
```
Comment (emmajane) [inline src/main/java/com/flowdock/jenkins/ChatMessage.java:107]: nice!

File: src/main/resources/com/flowdock/jenkins/FlowdockNotifier/config.jelly [hunk 3]
```diff
@@ -16,10 +21,18 @@
          title="Test connection with Chat notification" progress="Sending..."
          method="testConnection" with="flowToken,notificationTags" />
 
+      <f:entry title="Chat notification when build starts" field="chatStart">
```
Comment (emmajane) [inline src/main/resources/com/flowdock/jenkins/FlowdockNotifier/config.jelly]: 😍

**Review by emmajane — APPROVED** [review]
🕹 I'm sorry for being hateful and asking you to write shitty java things.

File: src/main/resources/com/flowdock/jenkins/FlowdockNotifier/config.jelly (hunk 3 above)
Comment (cafuego) [reply src/main/resources/com/flowdock/jenkins/FlowdockNotifier/config.jelly]: I mean, nice to maker a field on the config form, but it didn't actually do anything.

File: src/main/java/com/flowdock/jenkins/FlowdockBuildStart.java [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.Serializable;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.Optional;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import jenkins.YesNoMaybe;
+// import jenkins.tasks.SimpleBuildWrapper0;
```
Comment (cafuego) [inline src/main/java/com/flowdock/jenkins/FlowdockBuildStart.java]:
```suggestion
```

File: src/main/java/com/flowdock/jenkins/FlowdockNotifier.java [hunk 5]
```diff
@@ -25,11 +25,9 @@
 
 public class FlowdockNotifier extends Notifier {
 
-
     private final String flowToken;
     private final String flowThread;
     private final String notificationTags;
-    private final boolean chatStart;
```
Comment (cafuego) [inline src/main/java/com/flowdock/jenkins/FlowdockNotifier.java]: Yeah nah.

File: src/main/java/com/flowdock/jenkins/FlowdockNotifier.java [hunk 6]
```diff
@@ -44,13 +42,12 @@
     // Fields in config.jelly must match the parameter names in the "DataBoundConstructor"
     @DataBoundConstructor
     public FlowdockNotifier(String flowToken, String flowThread, String notificationTags,
-        String chatStart, String chatNotification,
```
Comment (cafuego) [inline src/main/java/com/flowdock/jenkins/FlowdockNotifier.java]: Yeah nah.

File: src/main/java/com/flowdock/jenkins/FlowdockNotifier.java [hunk 7]
```diff
@@ -44,13 +42,12 @@
...
+        String chatNotification,
         String chatStatus, String notifySuccess, String notifyFailure, String notifyFixed,
         String notifyUnstable, String notifyAborted, String notifyNotBuilt) {
         this.flowToken = flowToken;
         this.flowThread = flowThread;
         this.notificationTags = notificationTags;
-        this.chatStart = chatStart != null && chatStart.equals("true");
```
Comment (cafuego) [inline src/main/java/com/flowdock/jenkins/FlowdockNotifier.java]: Yeah nah.

File: src/main/resources/com/flowdock/jenkins/FlowdockNotifier/config.jelly [hunk 8]
```diff
@@ -21,10 +21,6 @@
          title="Test connection with Chat notification" progress="Sending..."
          method="testConnection" with="flowToken,notificationTags" />
 
-      <f:entry title="Chat notification when build starts" field="chatStart">
-        <f:checkbox />
-      </f:entry>
-
```
Comment (cafuego) [inline src/main/resources/com/flowdock/jenkins/FlowdockNotifier/config.jelly]: Yeah nah.

**Pull request merged**